configtx/channel.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package configtx
+
+import (
+	"fmt"
+
+	cb "github.com/hyperledger/fabric-protos-go/common"
+)
+
+// ChannelGroup encapsulates the parts of the config that control channels.
+// This type implements retrieval of the various channel config values.
+type ChannelGroup struct {
+	channelGroup *cb.ConfigGroup
+}
+
+// UpdatedChannelGroup is a ChannelGroup that can be modified in order to
+// generate a config update.
+type UpdatedChannelGroup struct {
+	*ChannelGroup
+}
+
+// Channel returns the channel group from the original config.
+func (o *OriginalConfig) Channel() *ChannelGroup {
+	return &ChannelGroup{channelGroup: o.ChannelGroup}
+}
+
+// Channel returns the channel group from the updated config.
+func (u *UpdatedConfig) Channel() *UpdatedChannelGroup {
+	return &UpdatedChannelGroup{ChannelGroup: &ChannelGroup{channelGroup: u.ChannelGroup}}
+}
+
+// Configuration returns a channel configuration value from a config transaction.
+func (c *ChannelGroup) Configuration() (Channel, error) {
+	var (
+		err          error
+		consortium   string
+		application  Application
+		orderer      Orderer
+		consortiums  []Consortium
+		capabilities []string
+	)
```

Comment:
Instead of 6 vars, should this simply be a Channel and err? Not sure the temporary is buying much since you're just populating the channel object.

Basically, assign each field after getting the data instead of carrying things. That keeps each of the field related blocks together. Given the length of the method, I think that would be easier to read.